core/common/src/main/java/alluxio/thread/AutopsyThread.java

```diff
@@ -54,9 +54,7 @@ public void setError(Throwable t) {
    *
    * @param t the crashing error
    */
-  public void onError(Throwable t) {
-    setError(t);
-  }
+  public abstract void onError(Throwable t);
```

Comment:
change this to abstract because I feel all implementations are different and embed their own reasons